front-end/src/components/CustomNode/NodeConfig.js

```diff
@@ -96,23 +118,49 @@ NodeConfig.propTypes = {
  */
 function OptionInput(props) {
 
+    const [isFlow, setIsFlow] = useState(props.flowValue ? true : false);
+
+    const handleFlowCheck = (bool) => {
+        // if un-checking, fire callback with null so no stale value is in `option_replace`
+        if (!bool) props.onChange(props.keyName, null, true);
+        setIsFlow(bool);
+    };
+
+    // fire callback to update `option_replace` with flow node info
+    const handleFlowVariable = (value) => {
+        props.onChange(props.keyName, value, true);
+    };
+
     let inputComp;
     if (props.type === "file") {
-        inputComp = <FileUploadInput {...props} />
+        inputComp = <FileUploadInput {...props} disabled={isFlow} />
     } else if (props.type === "string") {
-        inputComp = <SimpleInput {...props} type="text" />
+        inputComp = <SimpleInput {...props} type="text" disabled={isFlow} />
     } else if (props.type === "int") {
-        inputComp = <SimpleInput {...props} type="number" />
+        inputComp = <SimpleInput {...props} type="number" disabled={isFlow} />
     } else if (props.type === "boolean") {
-        inputComp = <BooleanInput {...props} />
+        inputComp = <BooleanInput {...props} disabled={isFlow} />
     } else {
         return (<></>)
     }
+
+    const hideFlow = props.node.options.node_type === "flow_control"
```

Comment:
KNIME allows for you to override flow node values with other flow nodes, so the first part here is not *entirely* correct. As-is though, I think this could cause an infinite loop on the server should a global flow node overwrite it's own value (it would keep looking up its own node). This *shouldn't* be a problem for local flow nodes (since they could never be provided as an option to themselves), so maybe change the check to `props.node.is_global`?